src/azul/project/hca/transformers.py

```diff
@@ -98,13 +102,16 @@ def __init__(self) -> None:
 
     def visit(self, entity: api.Entity) -> None:
         if isinstance(entity, api.SpecimenFromOrganism):
+            # This is to ensure that we have a unique set of specimens.
```

Comment:
I think these comments (`This is to ensure that we have a unique set of ...`) are unnecessary and should be removed. The primary purpose of these lines is to add the data to the visitor (and to do in a way that ensures uniqueness). This seems clear enough from the code.

Reply:
@mikebaumann, I think the comment is necessary.

As the way the code is designed, this method does not explicitly express the reason why we keep the data in a dictionary. It would require developers to look into the usage to get the idea why we need to collect the data in a dictionary, not in a list or set.

A few things I would prefer to have more explanation written/documented are:

* If the key is registered, why does the method always override the value (or the extracted version of the value)? Is it to simplify the code by not checking if the key already exists?
* Is there a chance that there are two slightly different value with the same key? If so, supposing that we choose to override the value for whatever reason, which version of value should we use?

I believe that in-code documentation will not help not just a newcomer, like myself, but other developers to understand what the code does and why it is designed to work this way.